packages/tracing/src/browser/request.ts

```diff
@@ -195,52 +195,88 @@ export function fetchCallback(
-    if (typeof headers.append === 'function') {
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-      headers.append('sentry-trace', span.toTraceparent());
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-      headers.append(BAGGAGE_HEADER_NAME, mergeAndSerializeBaggage(incomingBaggage, headers.get(BAGGAGE_HEADER_NAME)));
-    } else if (Array.isArray(headers)) {
-      const [, headerBaggageString] = headers.find(([key, _]) => key === BAGGAGE_HEADER_NAME);
-      headers = [
-        ...headers,
-        ['sentry-trace', span.toTraceparent()],
-        [BAGGAGE_HEADER_NAME, mergeAndSerializeBaggage(incomingBaggage, headerBaggageString)],
-      ];
-    } else {
-      headers = {
-        ...headers,
-        'sentry-trace': span.toTraceparent(),
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        baggage: mergeAndSerializeBaggage(incomingBaggage, headers.baggage),
-      };
+  if (!headers) {
+    return { 'sentry-trace': sentryTraceHeader, baggage: sentryBaggageHeader };
+  } else if (typeof Headers !== 'undefined' && isInstanceOf(headers, Headers)) {
+    const newHeaders = new Headers(headers as Headers);
+
+    newHeaders.append('sentry-trace', sentryTraceHeader);
+
+    if (sentryBaggageHeader) {
+      // If the same header is appended miultiple times the browser will merge the values into a single request header.
+      // Its therefore safe to simply push a "baggage" entry, even though there might already be another baggage header.
+      newHeaders.append(BAGGAGE_HEADER_NAME, sentryBaggageHeader);
     }
+
+    return newHeaders as PolymorphicRequestHeaders;
+  } else if (Array.isArray(headers)) {
+    const newHeaders = [...headers];
+    newHeaders.push(['sentry-trace', sentryTraceHeader]);
+
+    if (sentryBaggageHeader) {
+      // If there are multiple entries with the same key, the browser will merge the values into a single request header.
+      // Its therefore safe to simply push a "baggage" entry, even though there might already be another baggage header.
+      newHeaders.push([BAGGAGE_HEADER_NAME, sentryBaggageHeader]);
+    }
+
+    return newHeaders;
   } else {
-    headers = { 'sentry-trace': span.toTraceparent(), baggage: mergeAndSerializeBaggage(incomingBaggage) };
+    const existingBaggageHeader = 'baggage' in headers ? headers.baggage : undefined;
+    const newBaggageHeaders: string[] = [];
+
+    if (Array.isArray(existingBaggageHeader)) {
+      newBaggageHeaders.push(...existingBaggageHeader);
+    } else if (existingBaggageHeader) {
+      newBaggageHeaders.push(existingBaggageHeader);
+    }
+
+    if (sentryBaggageHeader) {
+      newBaggageHeaders.push(sentryBaggageHeader);
+    }
+
+    return {
+      ...(headers as Exclude<typeof headers, Headers>),
+      'sentry-trace': sentryTraceHeader,
+      baggage: newBaggageHeaders.length > 0 ? newBaggageHeaders.join(', ') : undefined,
```

Comment:
Why `', '` over `','`?

Reply:
No real reason tbh, changed to save a byte :D 18df790c51aed97aa1b3562c353d7cd16df4b1b0